### ISSUE TITLE
Scenario list glitches when you have too many scenarios

### ISSUE THREAD

**Issue by MarcelVos96** [OpenRCT2/OpenRCT2#12516]
**OS:** Windows 10
**Version:** 0.2.6
**Commit/Build:** bb1d814

If you have too many scenarios in your scenario list they become invisible once you scroll down far enough. It seems to happen after about 1350 scenarios. You can still select the scenarios and play them without any issues. When I tried adding enough scenarios to cause this issue in vanilla I lost the ability to scroll down at all so I don't know if this happens in vanilla too.

**Screenshot:**
![image](https://user-images.githubusercontent.com/25930360/88841687-94f12d00-d1de-11ea-808f-6f44fe779d9e.png)

**Comment by James103** [comment]
There's most likely a 16-bit overflow somewhere in the drawing code, since 1350 * 24 = 32400 pixels, which is very close to 32767.

**Comment by sjoerddebruin** [comment]
> There's most likely a 16-bit overflow somewhere in the drawing code, since 1350 * 24 = 32400 pixels, which is very close to 32767.

For the record, these are the related parts:

https://github.com/OpenRCT2/OpenRCT2/blob/252987e921fa07587f8192d3a72ac17719938c20/src/openrct2/drawing/Drawing.Sprite.cpp#L882

https://github.com/OpenRCT2/OpenRCT2/blob/252987e921fa07587f8192d3a72ac17719938c20/src/openrct2/drawing/Drawing.Sprite.cpp#L941

Most in that file already seem to have been widened to `int32_t` in #15112 and #19515. Are there performance reasons that hold this back as it's quite a shared function?

**Comment by Gymnasiast** [comment]
I don’t think so, so if you want to have a go at fixing this (and manage to acquire enough scenarios...) by all means go ahead!
